File: Sesion 7/Marco teorico/9herencia_protected/9.java
import java.util.*;

class Animal {
    protected int edad;

    public Animal(int edad) {
        this.edad = edad;
    }

    public void mostrarEdad() {
        System.out.println("Edad del animal: " + edad + " años");
    }
}

class Perro extends Animal {
    public Perro(int edad) {
        super(edad);
    }

    public void mostrarEdadDelPerro() {
        System.out.println("Edad del perro: " + edad + " años");
    }
}

public class Main {
    public static void main(String[] args) {
        Perro miPerro = new Perro(5);
        miPerro.mostrarEdadDelPerro();
    }
}

File: Sesion 7/Marco teorico/9herencia_protected/9.py
class Animal:
    def __init__(self, edad):
        self.edad = edad

    def mostrar_edad(self):
        print("Edad del animal:", self.edad, "años")

class Perro(Animal):
    def __init__(self, edad):
        super().__init__(edad)

    def mostrar_edad_del_perro(self):
        print("Edad del perro:", self.edad, "años")


mi_perro = Perro(5)
mi_perro.mostrar_edad_del_perro()
